### ISSUE TITLE
poetry install with nested packages with "--no-root-nested" or similar option to improve docker builds

### ISSUE THREAD

**Issue by yeus** [python-poetry/poetry#3671]
- [ x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Feature Request

Right now it is possible to install only the dependencies of a package with the `poetry install --no-root --no-dev` options. This is great to improve docker builds (cache behaviour for example).

it would be great to extend that functionality to nested packages. Something like

`poetry install --no-root --no-dev --no-root-nested`

which would tell poetry to only install the *dependencies* of the nested package, but
not the package itself. This would greatly improve the caching behaviour of docker builds as we would only need
to copy  the pyproject.toml and poetry.lock files into the docker machine which causes much better caching behaviour.

**Comment by killjoy1221** [comment]
A possible workaround could be done using `poetry export`. It's not ideal since it uses pip.

```dockerfile
# install dependencies, cached build
COPY pyproject.toml poetry.lock ./
RUN bash -c "pip install -r <(poetry export)" # optionally --without-hashes

# Actually install the app
COPY . .
RUN poetry install --no-dev
```

**Comment by yeus** [comment]
I have found a temporary workaround right now by only copying "pypoject.toml" and "poetry.lock" for each subproject. Additionally in order for this to work one needs a project subfolder with an __init__.py in it. After all dependencies got resolved and installed in docker I copy the rest of the files into the docker image. This way we get somewhat nice caching behaviour:

```dockerfile
# copy only the pyproject.toml and lockfile in order to make sure they are cached
# as graphengine depends on myprojectlibrary we also copy the myprojectlibrary pyproject.toml and poetry.lock directly
# to the dockerfile. 
COPY ./myproject/pyproject.toml ./myproject/poetry.lock "$PROJECT_DIR/myproject/"
COPY ./myprojectlibrary/pyproject.toml ./myprojectlibrary/poetry.lock "$PROJECT_DIR/myprojectlibrary/"
WORKDIR "$PROJECT_DIR/myproject"

# now we can do the installation without the actual packages which
# enables us to make use of the docker-cache. We will
# pull in the actual project files at a later stage
# In order for this to work we also need to create a "fake" myprojectlibrary directory
# we can install the project "normally" because we don't have the actual project files included here
RUN mkdir "$PROJECT_DIR/myprojectlibrary/comcharax_lib" &&\
    touch "$PROJECT_DIR/myprojectlibrary/comcharax_lib/__init__.py" &&\
    mkdir "$PROJECT_DIR/myproject/comcharax_graphengine" &&\
    touch "$PROJECT_DIR/myproject/comcharax_graphengine/__init__.py" &&\
    POETRY_VIRTUALENVS_IN_PROJECT=true poetry install --no-dev -vvv

COPY . $PROJECT_DIR
```

**Comment by yeus** [comment]
So given that by creating the mentioned directories works, it would probably be sufficient for poetry to simply ignore the library folder inside a python project when called with a flag like the one I mentioned above (`--no-root-nested`). (in my case "myprojectlibrary" and "myproject"). Those would normally get linked when installing subprojects. But that obviously isn't needed in this case ..

**Comment by neersighted** [comment]
What is being asked for right now is very unclear to me, but as best I can tell this is a sub-ask for #2270 -- if that is not correct, please update this issue with a clear explanation of what, from a environment/technical level, is being requested.

**Comment by adriangb** [comment]
This would be fixed exactly by #6845
